Document install, update, and delete procedures for Argo Rollouts (#61983)
openshift/openshift-docs: modules/gitops-about-argo-rollout-manager-custom-resources-and-spec.adoc

// Module is included in the following assemblies:
//

> // * openshift-docs/cicd/gitops/using-argo-rollouts-for-progressive-deployment-delivery.adoc

Reviewer: ```suggestion
// * cicd/gitops/using-argo-rollouts-for-progressive-deployment-delivery.adoc
```

:_content-type: CONCEPT
[id="gitops-about-argo-rollout-manager-custom-resources-and-spec_{context}"]
= About RolloutManager custom resources and specification

After you install {gitops-title} Operator on the cluster, to use the Argo Rollouts feature, you must create and submit RolloutManager custom resource (CR) with the required details to the {gitops-shortname} Operator in the namespace of your choice. But you can use Argo Rollouts in multiple namespaces as needed. The Operator creates an `argo-rollouts` instance with the following namespace-scoped supporting resources:

> 

Author: @iam-veeramalla  I tried installing Argo Rollouts with 1.9 cluster. I am unable to see this CR - argo-rollouts-manager-metrics-reader (ClusterRole).

I see the CR mentioned here in this doc: https://docs.google.com/document/d/1QlXHOidAvLFz10PwOMWmzXqlipqHES9EJr_-RticHyY/edit#heading=h.9vauumaztq62

Should I mention the argo-rollouts-manager-metrics-reader (ClusterRole) in this list? Please let me know. Thanks!

Author: Not required. @iam-veeramalla just confirmed.

* Argo Rollouts controller
* Argo Rollouts metrics service
* Argo Rollouts service account
* Argo Rollouts roles
* Argo Rollouts role bindings

> * Argo rollouts secret

Reviewer 2: ```suggestion
* Argo Rollouts secret
```

You can specify the command arguments, environment variables, a custom image name, and so on for the Argo Rollouts controller resource in the spec of the `RolloutsManager` CR. The `RolloutManager` CR spec defines the desired state of Argo Rollouts.

> .Example: RolloutManager CR

Author: Thanks for the suggestion @adellape. But the RolloutManager instances are UI-related and do not need backticks. Retaining them as is.

[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: RolloutManager
metadata:
  name: argo-rollout
  labels:
    example: basic
spec: {}
----

[id="argo-rollouts-controller_{context}"]
== Argo Rollouts controller

> With the Argo Rollouts controller resource, you can manage the progressive delivery of applications in the namespace of your choice. The Argo Rollouts controller resource monitors the cluster for events and reacts whenever there is a change in any Argo Rollouts-related resource. The controller reads all the rollout details and brings the cluster to the same state as described in the rollout definition.

Reviewer 3: ```suggestion
With the Argo Rollouts controller resource, you can manage progressive application delivery in your namespace. The Argo Rollouts controller resource monitors the cluster for events, and reacts whenever there is a change in any resource related to Argo Rollouts. The controller reads all the rollout details and brings the cluster to the same state as described in the rollout definition.
```
